### PR TITLE
Fix: solve metadata for Roles + TransactionRequest#children

### PR DESCRIPTION
**Pull request by 0xGabi**
Add a few small fixes from user feedback.

### CODE REVIEW

File: packages/connect-core/src/entities/Organization.ts [hunk 1]
```diff
@@ -105,6 +105,11 @@ export default class Organization {
     return this.#address || '' // The || '' should never happen but TypeScript requires it.
   }
 
+  get provider(): ethers.providers.Provider {
```
Comment (0xGabi) [inline packages/connect-core/src/entities/Organization.ts:108]: May be useful to have access to the same provider the connect library initialize.

File: packages/connect-core/src/entities/Role.ts [hunk 2]
```diff
@@ -31,7 +33,7 @@ export default class Role extends CoreEntity {
 
     const { roles } = metadata[0] as AragonArtifact
 
-    const role = roles.find(role => role.bytes === data.hash)
+    const role = roles?.find(role => role.bytes === data.hash)
```
Comment (0xGabi) [inline packages/connect-core/src/entities/Role.ts:36]: We had an error here when the `roles` were undefined.

File: packages/connect-thegraph/src/parsers/roles.ts [hunk 3]
```diff
@@ -33,8 +28,9 @@ async function _parseRole(
     manager: role.manager,
     hash: role.roleHash,
     grantees,
-    artifact,
-    contentUri,
+    appId: app.appId,
```
Comment (0xGabi) [inline packages/connect-thegraph/src/parsers/roles.ts:31]: Includes the `appId` on the `Role` entity so we can override the artifact metadata for the apps without a repo on aragonPM.

File: packages/connect-core/src/entities/App.ts [hunk 4]
```diff
@@ -78,24 +73,8 @@ export default class App extends CoreEntity {
...
-      )
-    }
-
-    const manifest: AragonManifest = await resolveMetadata(
-      'manifest.json',
-      data.contentUri || undefined,
-      data.manifest
-    )
+    const artifact = await resolveArtifact(data)
+    const manifest = await resolveManifest(data)
```
Comment (bpierre) [inline packages/connect-core/src/entities/App.ts:77]: ✨

File: packages/connect-core/src/entities/Organization.ts (hunk 1 above)
Comment (bpierre) [reply packages/connect-core/src/entities/Organization.ts:108]: Good point yes!

I think we could do this for now, and when making the “connection context” changes, move this inside a single `.connection` property that would contain the provider, the connector, etc. That way, we wouldn’t add too many properties on `Organization` that are in fact related to the connection.

File: packages/connect-core/src/entities/Role.ts (hunk 2 above)
Comment (bpierre) [reply packages/connect-core/src/entities/Role.ts:36]: Ah yes and we are using `as` before so TS couldn’t catch it :s

File: packages/connect-core/src/utils/descriptions.ts [hunk 5]
```diff
@@ -40,9 +40,9 @@ export async function describeTransaction(
         provider
       )
     }
-
-    // eslint-disable-next-line no-empty
-  } catch (_) {}
+  } catch (err) {
+    throw new Error(`Error: ${err}`)
```
Comment (bpierre) [inline packages/connect-core/src/utils/descriptions.ts:44]: Should we add a bit more context here?

```suggestion
    throw new Error(`Could not describe transaction: ${err}`)
```

File: packages/connect-core/src/utils/metadata.ts [hunk 6]
```diff
@@ -39,3 +48,29 @@ export async function resolveMetadata(
...
+    data.manifest
+  )
+}
+
+export async function resolveArtifact(
+  data: AppData | RoleData
+): Promise<AragonArtifact> {
+  if (hasAppInfo(data.appId, 'apm')) {
+    return getApmInternalAppInfo(data.appId)
+  } else if (hasAppInfo(data.appId, 'aragon')) {
```
Comment (bpierre) [inline packages/connect-core/src/utils/metadata.ts:67]: Now that we are returning from a function, we could get rid of the `else if` here, and the `else` below.

File: yarn.lock [hunk 7]
```diff
@@ -226,6 +226,16 @@
   dependencies:
```
Comment (bpierre) [inline yarn.lock:226]: Wondering if there was a reason to update the yarn.lock in this PR?

Comment (0xGabi) [reply yarn.lock:226]: No, I'll remove that.

File: packages/connect-core/src/utils/metadata.ts (hunk 6 above)
Comment (0xGabi) [reply packages/connect-core/src/utils/metadata.ts:67]: Nice, doing that change.

**Pull request merged**